test/smoke/testApps/SpringBootAuto/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/SpringSchedulerTest.java

```diff
@@ -0,0 +1,84 @@
+import com.google.common.base.Stopwatch;
+import com.microsoft.applicationinsights.internal.schemav2.Data;
+import com.microsoft.applicationinsights.internal.schemav2.Envelope;
+import com.microsoft.applicationinsights.internal.schemav2.RemoteDependencyData;
+import com.microsoft.applicationinsights.internal.schemav2.RequestData;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+@UseAgent
+public class SpringSchedulerTest extends AiSmokeTest {
+
+    @Test
+    @TargetUri("/scheduler")
+    public void fixedRateSchedulerTest() throws Exception {
+        Stopwatch stopwatch = Stopwatch.createStarted();
+        while (mockedIngestion.getCountForType("RequestData") < 20 && stopwatch.elapsed(TimeUnit.SECONDS) < 10) {
+        }
+
+        List<Envelope> rdList = mockedIngestion.getItemsEnvelopeDataType("RequestData");
+        assertTrue(rdList.size() >= 20);
+
+        List<Envelope> httpRequestList = new ArrayList<>(1);
+        List<Envelope> schedulerRequestList = new ArrayList<>();
+        groupRequestList(httpRequestList, schedulerRequestList, rdList);
+        assertEquals(1, httpRequestList.size());
+        assertTrue(schedulerRequestList.size() >= 20);
+
+        List<Envelope> rddList = mockedIngestion.getItemsEnvelopeDataType("RemoteDependencyData");
+
+        for (Envelope envelope : httpRequestList) {
+            RequestData rd = (RequestData) ((Data) envelope.getData()).getBaseData();
+            assertEquals("GET /scheduler", rd.getName());
+            Envelope rddEnvelope = findRemoteDependencyData(rd.getId(), new ArrayList<>(rddList));
+            if (rddEnvelope != null) {
+                RemoteDependencyData rdd = (RemoteDependencyData) ((Data) rddEnvelope.getData()).getBaseData();
+                assertEquals("TestController.scheduler", rdd.getName());
+                assertEquals(rd.getId(), rddEnvelope.getTags().get("ai.operation.parentId"));
+            }
+        }
+
+        for (Envelope envelope : schedulerRequestList) {
+            RequestData rd = (RequestData) ((Data) envelope.getData()).getBaseData();
+            assertEquals("SpringBootApp.fixedRateScheduler", rd.getName());
+            Envelope rddEnvelope = findRemoteDependencyData(rd.getId(), new ArrayList<>(rddList));
+            if (rddEnvelope != null) {
+                RemoteDependencyData rdd = (RemoteDependencyData) ((Data) rddEnvelope.getData()).getBaseData();
+                assertEquals("GET /search", rdd.getName());
+                assertEquals("www.bing.com", rdd.getTarget());
+                assertEquals(rd.getId(), rddEnvelope.getTags().get("ai.operation.parentId"));
+            }
+        }
+    }
+
+    private Envelope findRemoteDependencyData(String parentId, List<Envelope> rddList) {
+        for (Envelope envelope : rddList) {
+            if (envelope.getTags().get("ai.operation.parentId").equals(parentId)) {
+                return envelope;
+            }
+        }
+
+        return null;
```

Comment:
and remove null checks above, better to fail in tests if our assumptions are not met

```suggestion
        throw new IllegalStateException("no remote dependency found for parentId: " + parentId);
```

Reply:
but it will be null for certain cases though.. because I only wait for 20 count of RequestData.  DepdendencyData is not guaranteed though. sometimes, dependencyData for a request is not part of the list yet.

Reply:
let's wait for 20 DependencyData also, then we can confirm everything

Reply:
I have tried hard, it's not possible to get all 20 dependency data that maps to the same 20 request.  We have dependency data from Message Data too.  the list i have will miss one or two.  waitForItems on either type failed for me.  there is no other option.